src/rpc/ngHandlers/Unsubscribe.h

```diff
@@ -0,0 +1,212 @@
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#pragma once
+
+#include <backend/BackendInterface.h>
+#include <rpc/RPCHelpers.h>
+#include <rpc/common/Types.h>
+#include <rpc/common/Validators.h>
+
+namespace RPCng {
+
+template <typename SubscriptionManagerType>
+class BaseUnsubscribeHandler
+{
+    std::shared_ptr<BackendInterface> sharedPtrBackend_;
+    std::shared_ptr<SubscriptionManagerType> subscriptions_;
+
+public:
+    struct OrderBook
+    {
+        ripple::Book book;
+        bool both = false;
+    };
+
+    struct Input
+    {
+        std::optional<std::vector<std::string>> accounts;
+        std::optional<std::vector<std::string>> streams;
+        std::optional<std::vector<std::string>> accountsProposed;
+        std::optional<std::vector<OrderBook>> books;
+    };
+
+    using Output = VoidOutput;
+    using Result = RPCng::HandlerReturnType<Output>;
+
+    BaseUnsubscribeHandler(
+        std::shared_ptr<BackendInterface> const& sharedPtrBackend,
+        std::shared_ptr<SubscriptionManagerType> const& subscriptions)
+        : sharedPtrBackend_(sharedPtrBackend), subscriptions_(subscriptions)
+    {
+    }
+
+    RpcSpecConstRef
+    spec() const
+    {
+        static auto const booksValidator =
+            validation::CustomValidator{[](boost::json::value const& value, std::string_view key) -> MaybeError {
+                if (!value.is_array())
+                    return Error{RPC::Status{RPC::RippledError::rpcINVALID_PARAMS, std::string(key) + "NotArray"}};
+                for (auto const& book : value.as_array())
+                {
+                    if (!book.is_object())
+                        return Error{
+                            RPC::Status{RPC::RippledError::rpcINVALID_PARAMS, std::string(key) + "ItemNotObject"}};
+                    if (book.as_object().contains("both") && !book.as_object().at("both").is_bool())
+                    {
+                        return Error{RPC::Status{RPC::RippledError::rpcINVALID_PARAMS, "bothNotBool"}};
+                    }
+                    auto const parsedBook = RPC::parseBook(book.as_object());
+                    if (auto const status = std::get_if<RPC::Status>(&parsedBook))
+                        return Error(*status);
+                }
+                return MaybeError{};
+            }};
+
+        static auto const rpcSpec = RpcSpec{
+            {JS(streams), validation::SubscribeStreamValidator},
+            {JS(accounts), validation::SubscribeAccountsValidator},
+            {JS(accounts_proposed), validation::SubscribeAccountsValidator},
+            {JS(books), booksValidator}};
+
+        return rpcSpec;
+    }
+
+    Result
+    process(Input input, Context const& ctx) const
+    {
+        if (input.streams)
+        {
```

Comment:
I'd omit braces here too, especially now that we have more characters per column and it won't look bad if you omit the braces